app/views/unattended/provisioning_templates/snippet/preseed_netplan_generic_interface.erb

```diff
@@ -17,24 +17,24 @@ oses:
         dhcp4: <%= @dhcp %>
         dhcp6: <%= @dhcp6 %>
 <%- if !@dhcp && !@subnet.nil? && !@interface.ip.nil? -%>
-        addresses: [ <%= @interface.ip %>/<%= @subnet.cidr %> ]
+        addresses:
```

Comment:
This does assume dual stack. This header will be missing on IPv6-only. I'd suggest something like
```erb
<%-
static_v4 = !@dhcp && !@subnet.nil? && !@interface.ip.nil?
static_v6 = !@dhcp6 && !@subnet6.nil? && !@interface.ip6.nil?
-%>
<%- if static_v4 || static_v6 %>
        addresses:
<%- end %>
<%- if static_v4 %>
          - <%= @interface.ip %>/<%= @subnet.cidr %>
<%- end %>
<%- if static_v6 %>
          - <%= @interface.ip6 %>/<%= @subnet6.cidr %>
<%- end %>
```

Later on you can also reuse the variables to set `gateway4` and `gateway6`.

Reply:
agreed - this is tidyer too - now done for the overall addresses block

for gateways i figured safer to leave it as close to original, i am guessing there might be wierd edge cases where a gateway is not provided for a static v6 or v4 (who knows) and its least change

can easily change it later i guess

Reply:
Thanks. It looks better.

> for gateways i figured safer to leave it as close to original, i am guessing there might be wierd edge cases where a gateway is not provided for a static v6 or v4 (who knows) and its least change

I think you're deviating a bit too much and suggested inline how to keep it the same as it was.